src/manifold/src/impl.cpp

```diff
@@ -478,26 +478,35 @@ Manifold::Impl::Impl(const Mesh& mesh, const MeshRelationD& relation,
   Finish();
 }
 
-void Manifold::Impl::Hull(const std::vector<glm::vec3>& vertPos){
-  int numVert = vertPos.size();
-  if (numVert < 4) return; // What to return here? 
+void Manifold::Impl::Hull(const std::vector<glm::vec3>& vertPos) {
+  size_t numVert = vertPos.size();
+  if (numVert < 4) {
+    status_ = Error::InvalidConstruction;
+    return;
+  }
 
   Vec<glm::dvec3> pointCloudVec(numVert);
-  for (int i = 0; i < numVert; i++) {
-    pointCloudVec[i] = {vertPos[i].x, vertPos[i].y, vertPos[i].z};
-  }
+  manifold::transform(vertPos.begin(), vertPos.end(), pointCloudVec.begin(),
+                      [](const glm::vec3& v) { return glm::dvec3(v); });
   QuickHull qh(pointCloudVec);
-  std::pair<Vec<Halfedge>, Vec<glm::dvec3>> retVal;
-  retVal= qh.getConvexHullAsMesh(pointCloudVec, false, true);
-  for (size_t i=0;i<retVal.second.size();i++){
-    vertPos_.push_back(glm::vec3(retVal.second[i].x, retVal.second[i].y, retVal.second[i].z));
-  }
-  numVert=vertPos_.size();
-  halfedge_=retVal.first;
+  ConvexHull hull = qh.getConvexHullAsMesh(pointCloudVec, false);
+  vertPos_.resize(hull.vertices.size());
+  manifold::transform(hull.vertices.begin(), hull.vertices.end(),
+                      vertPos_.begin(),
+                      [](const glm::dvec3& v) { return glm::vec3(v); });
```

Comment:
Let's make a note that we can replace with this `move` once our `double`s PR lands.